src/nvim/os/os_defs.h

```diff
@@ -1,10 +1,145 @@
 #ifndef NVIM_OS_OS_DEFS_H
 #define NVIM_OS_OS_DEFS_H
 
+#include <ctype.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <sys/stat.h>
+#include <sys/types.h>
+
 #ifdef WIN32
 # include "nvim/os/win_defs.h"
 #else
 # include "nvim/os/unix_defs.h"
 #endif
 
+#ifdef SIGHAS3ARGS
```

Comment:
@equalsraf these aren't defined anywhere and haven't been touched since the initial import. Do you know if they're needed on Windows? Same goes for `RETSIGTYPE` and `SIGDEFARG`, whose semantics are apparently specified in C89 and above.